### ISSUE TITLE
Kafka Health Check cannot be disabled

### ISSUE THREAD

**Issue by erard22** [quarkusio/quarkus#13920]
**Describe the bug**

In contrast to the documentation it seems the Kafka Health Check is not disabled by default. Even setting the proprety `quarkus.kafka.health.enabled=false` does not disable it.

**Expected behavior**

That the behaviour is consistent with the docuementation and the check can be disabled.

**Actual behavior**

I found no way to disable the check. This is important because my application has an online part and a async part, where we are processing kafka messages. The online part should stay online, event if there was a failing kafka message.

**To Reproduce**

I did a little demo project to reproduce the behaviour: 

https://github.com/erard22/kafka-health-test

- `mvn docker:build docker:start`
- `mvn quarkus:dev`
- Run the test `DemoConsumerIT.java`
- Go to http://localhost:8080/health

**Configuration**
The current configuration  of the demo project should disable the health check on both ways, Quarkus and Smallrye: 

```
mp.messaging.incoming.demo.health-enabled=false
mp.messaging.incoming.demo.health-readiness-enabled=false
quarkus.kafka.health.enabled=false
```

**Output**

```json
{
    "status": "DOWN",
    "checks": [
        {
            "name": "SmallRye Reactive Messaging - liveness check",
            "status": "DOWN",
            "data": {
                "application-ch.erard22.quarkus.DemoConsumer#consume": "[KO] - failed to process: Testmessage"
            }
        },
        {
            "name": "SmallRye Reactive Messaging - readiness check",
            "status": "DOWN",
            "data": {
                "application-ch.erard22.quarkus.DemoConsumer#consume": "[KO] - failed to process: Testmessage"
            }
        }
    ]
}
```

**Additional context**

For the Smallrye properties it sees to be working. Changing them to 

```
mp.messaging.incoming.demo.health-enabled=true
mp.messaging.incoming.demo.health-readiness-enabled=true
quarkus.kafka.health.enabled=false
```

adds additional information to the health output, provided by Smallrye: 

```json
{
    "status": "DOWN",
    "checks": [
        {
            "name": "SmallRye Reactive Messaging - liveness check",
            "status": "DOWN",
            "data": {
                "application-ch.erard22.quarkus.DemoConsumer#consume": "[KO] - failed to process: Testmessage",
                "demo": "[KO] - failed to process: Testmessage"
            }
        },
        {
            "name": "SmallRye Reactive Messaging - readiness check",
            "status": "DOWN",
            "data": {
                "application-ch.erard22.quarkus.DemoConsumer#consume": "[KO] - failed to process: Testmessage",
                "demo": "[OK]"
            }
        }
    ]
}
```

**Comment by xstefank** [comment]
This seems specific to reactive messaging so can you try `quarkus.reactive-messaging.health.enabled`? Otherwise, I'll leave this to @cescoffier.

**Comment by erard22** [comment]
You're right. Didn't find this in the specification, but it seems it's doing the job. Thanks!

**Comment by effusion** [comment]
Hey Guys,

I'm currently facing an issue with the health check. We have a Kafka ingest on which we receive JSON Message. Sometimes we get a malformed JSON and the Custom JacksonDesieralizer

`class JsonDocumentDeserializer : ObjectMapperDeserializer<DocumentV12>(DocumentV12::class.java)`

throws an exception turning the channel to KO and therefore the entire health to down.

I did try to disable the health check for this channel with `mp.messaging.incoming.srf-cms-ingest-v12.health-enabled=false` but it gets reported anyway.
``"application-ch.srgssr.pdp.api.ingest.consumer.DocumentConsumer#receiveFromCMS": "[KO] - This consumer has already been closed."`
`
The only thing that helped was to disable the reactive-health completely with `quarkus.reactive-messaging.health.enabled=false`

Hope this is enough information
